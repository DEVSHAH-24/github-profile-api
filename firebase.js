import { getAuth } from "firebase/auth";
import { initializeApp } from "firebase/app";

const firebaseConfig = {
  apiKey: "AIzaSyClzkXRH_mOI-WyKvF7h3cSyQ-CLutaI7c",
  authDomain: "github-api-66ce7.firebaseapp.com",
  projectId: "github-api-66ce7",
  storageBucket: "github-api-66ce7.appspot.com",
  messagingSenderId: "708508000411",
  appId: "1:708508000411:web:47c652ef87701f9289e63e",
};
let app;
app = initializeApp(firebaseConfig);
export const auth = getAuth(app);
