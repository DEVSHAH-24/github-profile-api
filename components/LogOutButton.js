import { useNavigation } from "@react-navigation/native";
import { signOut } from "firebase/auth";
import React from "react";
import { Button } from "react-native";
import { auth } from "../firebase";

export const LogOutButton = () => {
  const navigation = useNavigation();

  const handleSignOut = () => {
    signOut(auth)
      .then(() => {
        navigation.navigate("Login");
        console.log(auth.currentUser);
      })
      .catch((e) => alert(e.message));
  };
  return (
    <Button testID="logout-btn" title="Logout" onPress={handleSignOut}></Button>
  );
};
export default LogOutButton;
