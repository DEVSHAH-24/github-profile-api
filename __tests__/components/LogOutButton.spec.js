import { useNavigation } from "@react-navigation/native";
import { signOut } from "firebase/auth";
import * as React from "react";
import TestRenderer from "react-test-renderer";
import LogOutButton from "../../components/LogOutButton";

const getComponent = (props = {}) => <LogOutButton {...props} />;
const mockedUsedNavigate = jest.fn();

jest.mock("firebase/auth", () => ({ signOut: jest.fn() }));

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,

    useNavigation: () => ({
      navigation: mockedUsedNavigate,
    }),
  };
});
describe("LogOutButton", () => {
  beforeEach(() => {
    mockedUsedNavigate.mockReset();
  });
  test("snapshot", () => {
    const navigation = useNavigation();

    const component = getComponent();

    const testRenderer = TestRenderer.create(component);

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  test("handle sign out", () => {
    const logOutProps = {
      handleSignOut: () => {},
    };

    const navigation = useNavigation();

    const component = getComponent();

    signOut.mockImplementation((auth) => auth.signOut());

    const testRenderer = TestRenderer.create(component);

    // const handleSignOut = () => {
    //   signOut(auth)
    //     .then(() => {
    //       navigation.navigate("Login");
    //     })
    //     .catch((e) => alert("hello"));
    // };
    // expect(testRenderer.toJSON()).toContain(handleSignOut());
  });
});
