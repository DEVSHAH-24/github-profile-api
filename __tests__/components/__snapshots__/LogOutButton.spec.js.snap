// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogOutButton button on press 1`] = `<LogOutButton />`;
